carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -509,7 +509,15 @@ void CARMAWorldModel::setMap(lanelet::LaneletMapPtr map, size_t map_version, boo
 
     ROS_INFO_STREAM("Building routing graph");
 
-    TrafficRulesConstPtr traffic_rules = *(getTrafficRules(lanelet::Participants::Vehicle));
+    TrafficRulesConstPtr traffic_rules;
+    if(participant_type_ == lanelet::Participants::VehicleCar)
+    {
+      traffic_rules = *(getTrafficRules(lanelet::Participants::VehicleCar));
+    }
+    else if(participant_type_ == lanelet::Participants::VehicleTruck)
+    {
+      traffic_rules = *(getTrafficRules(lanelet::Participants::VehicleTruck));
+    }
```

Comment:
What is the point for these if checks? Seems like this whole block is equivalent to the one liner 
```
traffic_rules = *(getTrafficRules(participant_type_ ));
```